source/Irrlicht/CBAWMeshFileLoader.cpp

```diff
@@ -210,19 +210,20 @@ ICPUMesh* CBAWMeshFileLoader::createMesh(io::IReadFile* _file)
 	const uint32_t filesize = _file->getSize();
 	uint8_t* const fileBuffer = (uint8_t*)malloc(filesize);
 	_file->read(fileBuffer, filesize);
```

Comment:
No need to read the entire file at once.

You dont check the file header, it needs to be done in order not to load some bogus file and crash.

First only read the file header and check that it even matches, and determine the file version.... then dispatch to the correct loadHeadersVN function.